prov/usnic/src/usdf_cm.c

```diff
@@ -634,31 +672,59 @@ int usdf_cm_dgram_getname(fid_t fid, void *addr, size_t *addrlen)
 		assert(sin.sin_addr.s_addr ==
 			ep->ep_domain->dom_fabric->fab_dev_attrs->uda_ipaddr_be);
 	}
-	memcpy(addr, &sin, copylen);
 
-	if (copylen < sizeof(sin))
-		return -FI_ETOOSMALL;
-	else
-		return 0;
+	if (addr_format_str) {
+		actual_addr_size = sizeof(addr_str);
+		usdf_addr_tostr(&sin, addr_str, &actual_addr_size);
+		memcpy(addr, addr_str, MIN(*addrlen, actual_addr_size));
```

Comment:
You should make sure that you add a `NULL` terminator in case the string is truncated. Also, add a test case for this.